### DISCUSSION TITLE
Control exhaust vent's flow by CO volume fraction

### DISCUSSION THREAD

**Discussion by cunconbkhp** [firemodels/fds#12829] [General]
Hello everyone!

I have a simple case test of a room fire where the volume flow of the smoke evcuation vent is controlled  by the monoxide carbon 's volume fraction monitored at a reference point inside the room. the fan will run at half speed when the concentration of CO is more than 25 ppm (0.000025 mol/mol) and at full speed when the CO concentration is more than 50 ppm (0.00005 mol/mol). 

&SURF ID='EXT',
      RGB=26,128,26,
      VOLUME_FLOW=1.0,
      RAMP_V='EXT_RAMP_V'/
&RAMP ID='EXT_RAMP_V', T=0.0, F=0.0, DEVC_ID='CO_FV_0'/
&RAMP ID='EXT_RAMP_V', T=2.4E-5, F=0.0/
&RAMP ID='EXT_RAMP_V', T=2.5E-5, F=0.5/
&RAMP ID='EXT_RAMP_V', T=4.9E-5, F=0.5/
&RAMP ID='EXT_RAMP_V', T=5.0E-5, F=1.0/
&RAMP ID='EXT_RAMP_V', T=1.0, F=1.0/

The result show a delay of the volume flow compared to the temporal evolution of the CO device.  As we can see : the under limit of 25ppm is well reached at about t = 40 s and the upper limit is reached at t = 50 s. However the volume flow curve is still zero until t = 60 s. At another bigger simulation, we have even more delay (about 100 s) beetween the CO device and the volume flow.
   
![image](https://github.com/firemodels/fds/assets/14924739/b9bab6f2-eeac-4eef-b301-abdd09b954c6)

Hope some one can give me some insights to better understand this result. Thank you in advance.

the fds text file and the result plot are placed below 
[test_CO_mode_R2.fds.txt](https://github.com/firemodels/fds/files/15031130/test_CO_mode_R2.fds.txt)
![result](https://github.com/firemodels/fds/assets/14924739/fa35a577-7294-4cf1-a16c-f6b4ecf9bca6)

**Comment by drjfloyd** [comment]
To make evaluating ramps efficient, FDS takes the T,F values given and creates a uniformly spaced array between the lowest T and the highest T with NUMBER_INTERPOLATION_POINTS in the array.  If FDS just used the T,F as given (which are not required to be uniformly spaced) it would have to search through the array to find the T value closest to the current ramp input. With the T,F mapped to a uniformly spaced array it is easy to compute what array value to choose.  In your case with T=1 and the default N_I_P of 5000, your points are spaced at 2E-4. As a result all the low values of the RAMP are too low.  Just delete the last line with T=1. Now your spacing will be 1E-8. You actually also do not need the first line for T=0,F=0 as you also have T=2.45-E5,F=0.  For input values outside the bounds of the RAMP, FDS just uses the value at the bounds.

**Comment by drjfloyd** [comment]
Though if you do also remove the T=0 line make sure to put the DEVC_ID on the next line.

**Reply by cunconbkhp** [reply]
Hi drjfloyif, 

i've deleted the last line and it works line a charm.

Thank you very much!

**Comment by Selvahasan** [comment]
Sir  
 I am working on car parking ventilation cfd I don't know how to do simulation send me related video I don't know how to do settings
Please send me a video and a project on how to do the simulation

**Comment by Selvahasan** [comment]
My mail id selvahasan@gmail.com

**Reply by drjfloyd** [reply]
It is not our role to do your work. It takes time to learn to use FDS and you need to spend the time to do so. Read the User's Guide,  Technical Reference, Verification Guide, and Validation. All of the inputs used in the Guides are available to you either in the Examples folder where you installed FDS or in the Validaiton directory on Github.

If this is for a real world project where people's lives will depend on your a ability to properly use FDS think carefully if this is work you should be doing given your current level of skill with FDS.
